packages/dd-trace/test/appsec/passport.spec.js

```diff
@@ -0,0 +1,239 @@
+'use strict'
+
+const proxyquire = require('proxyquire')
+
+describe('Passport', () => {
+  const rootSpan = {
+    context: () => { return {} }
+  }
+  const loginLocal = { type: 'local', username: 'test' }
+  const userUuid = {
+    id: '591dc126-8431-4d0f-9509-b23318d3dce4',
+    email: 'testUser@test.com',
+    username: 'Test User'
+  }
+
+  let passportModule, log, events, setUser
+  beforeEach(() => {
+    log = {
+      warn: sinon.stub()
+    }
+
+    events = {
+      trackEvent: sinon.stub()
+    }
+
+    setUser = {
+      setUserTags: sinon.stub()
+    }
+
+    passportModule = proxyquire('../../src/appsec/passport', {
+      '../log': log,
+      './sdk/track_event': events,
+      './sdk/set_user': setUser
+    })
+  })
+
+  describe('passportTrackEvent', () => {
+    it('should call log when credentials is undefined', () => {
+      passportModule.passportTrackEvent(undefined, undefined, undefined, 'safe')
+
+      expect(log.warn).to.have.been.calledOnceWithExactly('No username found in authentication instrumentation')
+    })
+
+    it('should call log when type is not known', () => {
+      const credentials = { type: 'unknown', username: 'test' }
+
+      passportModule.passportTrackEvent(credentials, undefined, undefined, 'safe')
+
+      expect(log.warn).to.have.been.calledOnceWithExactly('No username found in authentication instrumentation')
+    })
+
+    it('should call log when type is known but username not present', () => {
+      const credentials = { type: 'unknown' }
```

Comment:
bad copy paste no ? it says the type is known in the title of the test
```suggestion
      const credentials = { type: 'http' }
```